internal/testutils/seed.go

```diff
@@ -0,0 +1,25 @@
+package testutils
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+type lockedCustomSource struct {
+	lk sync.Mutex
+	s  rand.Source
+}
+
+func Seed() {
+	src := lockedCustomSource{
```

Comment:
Yeah so as discussed on Slack, I used `sync.Do()` to handle printing it once during tests and created a helper which returns the `Source` for each dedicated test to avoid using a global `Source`.